core/iwasm/common/wasm_shared_memory.c

```diff
@@ -176,7 +176,7 @@ acquire_wait_info(void *address, AtomicWaitNode *wait_node)
     AtomicWaitInfo *wait_info = NULL;
     bh_list_status ret;
 
-    if (address)
+    if (wait_map && address) /* avoid passing NULL to bh_hash_map_find */
```

Comment:
as far as i understand, "address" here can't be NULL.

Reply:
Yes, you are right, here the address comes from opcode atomic.wait32, atomic.wait64 and atomic.notify, all of them operate on an address inside the linear memory, which should not be NULL. I changed to bh_assert instead.